src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -1393,6 +1394,16 @@ public int executeCountQuery(String query) throws SQLException {
     }
   }
 
+  // WARNING: SQL string must be sanitized before calling this method
+  public <T> List<T> runQuery(String sql, Function<ResultSet, T> aggregateResult) {
+    try {
+      return synapseJdbcTemplate.query(sql, (rs, rowNum) -> aggregateResult.apply(rs));
+    } catch (DataAccessException ex) {
+      logger.warn(EMPTY_TABLE_ERROR_MESSAGE, ex);
+      return new ArrayList<>();
```

Comment:
It's pretty much the same but I would use `List.of()` instead since the result is immutable.
```suggestion
      return List.of();
```

Reply:
Actually, the code wouldn't build with this change. I wonder if it's because the return type is generic?

Reply:
I wonder why it didn't work; I tried changing it in the IDE and it didn't generate an error. Maybe it was a spot bugs finding?